redfish/collector.go

```diff
@@ -0,0 +1,241 @@
+package redfish
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/cybozu-go/log"
+	"github.com/cybozu-go/setup-hw/config"
+	"github.com/cybozu-go/setup-hw/gabs"
+	"github.com/prometheus/client_golang/prometheus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const namespace = "hw"
+
+// Collector implements prometheus.Collector interface.
+type Collector struct {
```

Comment:
You can make this non-exported because the return value of `NewCollector` can be `prometheus.Collector`.